aspnetcore/fundamentals/hosting.md

```diff
@@ -1,225 +1,534 @@
+**Set using**: `UseSetting`
 
-**Environment** `string`
+This feature is new in ASP.NET Core 2.0.
 
-Key: `environment`. Defaults to "Production". May be set to any value. Framework-defined values include "Development", "Staging", and "Production". Values are not case sensitive. See [Working with Multiple Environments](environments.md). Set using the `UseEnvironment` method.
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
 ```csharp
-new WebHostBuilder()
-    .UseEnvironment("Development")
+WebHost.CreateDefaultBuilder(args)
+    .UseSetting(WebHostDefaults.PreventHostingStartupKey, "true")
+    ...
 ```
 
-> [!NOTE]
-> By default, the environment is read from the `ASPNETCORE_ENVIRONMENT` environment variable. When using Visual Studio, environment variables may be set in the *launchSettings.json* file.
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
+
+```csharp
+var host = new WebHostBuilder()
+    .UseSetting(WebHostDefaults.PreventHostingStartupKey, "true")
+    ...
+```
+
+---
+
+### Server URLs
+
+Indicates the IP addresses or host addresses with ports and protocols that the server should listen on for requests.
+
+**Key**: urls  
+**Type**: *string*  
+**Default**: http://localhost:5000  
+**Set using**: `UseUrls`
+
+Set to a semicolon-separated (;) list of URL prefixes to which the server should respond. For example, `http://localhost:123`. Use "\*" to indicate that the server should listen for requests on any IP address or hostname using the specified port and protocol (for example, `http://*:5000`). The protocol (`http://` or `https://`) must be included with each URL. Supported formats vary between servers.
+
+# [ASP.NET Core 2.x](#tab/aspnetcore2x)
 
-<a id="server-urls"></a>
+```csharp
+WebHost.CreateDefaultBuilder(args)
+    .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+    ...
+```
 
-**Server URLs** `string`
+Kestrel has its own endpoint configuration API. For more information, see [Kestrel web server implementation in ASP.NET Core](xref:fundamentals/servers/kestrel?tabs=aspnetcore2x#endpoint-configuration).
 
-Key: `urls`. Set to a semicolon (;) separated list of URL prefixes to which the server should respond. For example, `http://localhost:123`. The domain/host name can be replaced with "\*" to indicate the server should listen to requests on any IP address or host using the specified port and protocol (for example, `http://*:5000` or `https://*:5001`). The protocol (`http://` or `https://`) must be included with each URL. The prefixes are interpreted by the configured server; supported formats will vary between servers.
+# [ASP.NET Core 1.x](#tab/aspnetcore1x)
 
 ```csharp
-new WebHostBuilder()
+var host = new WebHostBuilder()
     .UseUrls("http://*:5000;http://localhost:5001;https://hostname:5002")
+    ...
+```
+
+---
+
+### Shutdown Timeout
+
+Specifies the amount of time to wait for the web host to shutdown.
+
+**Key**: shutdownTimeoutSeconds  
+**Type**: *int*  
+**Default**: 5  
+**Set using**: `UseSetting`
```

Comment:
There is also an extension for this: https://github.com/aspnet/Hosting/blob/rel/2.0.0/src/Microsoft.AspNetCore.Hosting.Abstractions/HostingAbstractionsWebHostBuilderExtensions.cs#L168